src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/MethodInfoExtensions.cs

```diff
@@ -9,13 +9,65 @@ public static MethodInfo GetUnderlyingGenericTypeMethod(this MethodInfo construc
         {
             var constructedType = constructedTypeMethod.DeclaringType;
             var genericTypeDefinition = constructedType.GetGenericTypeDefinition();
+            var genericArguments = constructedType.GenericTypeArguments;
 
             // Retrieve list of candidate methods that match name and parameter count
             var candidateMethods = genericTypeDefinition.GetMethods()
                 .Where(m =>
```

Comment:
This method has gotten pretty complicated - feels like it could use with some dedicated unit tests, rather than just relying on the (non-exhaustive) verification tests.

Reply:
Yeah.. If you see the logic, it's pretty much the logic that was previously with the exception that right now you distinguish between two methods with the same number of parameters, but different kind. I'll do a UnitTest, but even for this simple logic the UnitTest is going to be hard

Reply:
Done!